### PR TITLE
Previous search intent

### LINKED ISSUE #562: "Previous result" command makes a new Google search
### Prerequisites:
Mic permissions are enabled.
A "Search lion" command is made to have the result in the "Doorhanger" and two "Next result" commands are made consecutively to have the new tab opened with the second website from the Google search list.

### STR:
1. Use the shortcut or click on the mic icon from the browser toolbar.
2. Say or write a "Previous result" command.

### Expected result:
The previous website is displayed in the tab regarding lions.

### Actual result:
A new Google search is made for "previous result" and the first website from the search list opens in a new tab.

### Notes:
Reproduced on Mac 10.14.6 and Win10x64 with Firefox Nightly 72.0a1 (64-bit).

### PR DESCRIPTION
**Pull request by maitrella**
New previous search intent and fix conflict with previous music intent.
Trying to fix #562 one more time.

### CODE REVIEW

File: extension/intents/music/music.js [hunk 1]
```diff
@@ -154,7 +154,7 @@ this.intents.music = (function() {
     play next (song | track |)        [direction=next]
     next (song | track)             [direction=next]
     play previous (song | track |)    [direction=back]
-    previous (song | track |)         [direction=back]
+    previous (song | track)         [direction=back]
```
Comment (ianb) [inline extension/intents/music/music.js]: Good catch, I messed up on this one with "next" earlier

File: extension/intents/search/search.js [hunk 2]
```diff
@@ -125,6 +125,59 @@ this.intents.search = (function() {
...
 
+  async function moveResult(context, step) {
+    stopCardPoll();
+    const { tabId, searchInfo } = await getSearchInfo();
+    if (!searchInfo) {
+      const e = new Error("No search made");
+      e.displayMessage = "You haven't made a search";
+      throw e;
+    }
+    if ((searchInfo.index >= searchInfo.searchResults.length - 1 && step > 0) ||
```
Comment (ianb) [inline extension/intents/search/search.js]: Looks like there's some formatting issues here. Did you run `npm run format` ?

File: extension/intents/search/search.js [hunk 3]
```diff
@@ -125,6 +125,59 @@ this.intents.search = (function() {
...
+    (searchInfo.index <= 0 && step < 0)) {
+      const tabId = await openSearchTab();
+      await context.makeTabActive(tabId);
+      return;
+    }
+    searchInfo.index += step;
```
Comment (ianb) [inline extension/intents/search/search.js]: This should probably test that `searchInfo.index + step >= 0`, and give an error if it's not.

File: extension/intents/search/search.js (hunk 2 above)
Comment (maitrella) [reply extension/intents/search/search.js]: Only eslint integration in Atom editor (0 errors) and npm install & npm start to test the code (following the README).Sorry I'll do npm run format before any commit.

File: extension/intents/search/search.js (hunk 3 above)
Comment (maitrella) [reply extension/intents/search/search.js]: I'll test that condition before increment line. What error message do you expect?.

I need some step by step guidance in order to update this PR with changes requested. I dont want to create another "rebase" mess.

Comment (ianb) [reply extension/intents/search/search.js]: I guess the error can be "No previous search result" (using the whole `exc = new Error("No previous search result"); exc.displayMessage = "No previous search result"; throw exc` pattern)

You can just push a new commit. I'll squash it in GitHub when it lands. 

Another easy way to revise a commit is `git commit -a --amend` (this adds your changes to the previous commit) and then `git push -f`

Comment (maitrella) [reply extension/intents/search/search.js]: Thanks!

**Comment by ianb** [comment]
Thanks!

**Pull request merged**